### PR TITLE
fix farm ut failed

### PR DESCRIPTION
**Pull request by Frank2333333**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/farm/handler_test.go [hunk 1]
```diff
@@ -45,7 +48,7 @@ type testCaseItem struct {
 	preExec      preExecFunc      // function "preExec" executes the code before executing the specific handler to be tested
 	getMsg       getMsgFunc       // function "getMsg" returns a sdk.Msg for testing, this msg will be tested by executing the function "handler"
 	verification verificationFunc // function "verification" Verifies that the test results are the same as expected
-	expectedCode uint32     // expectedCode represents the expected code in the test result
+	expectedCode sdk.EnvelopedErr     // expectedCode represents the expected code in the test result
```
Comment (ilovers) [inline x/farm/handler_test.go]: better to change expectedCode to expectedErr.

File: x/farm/handler_test.go [hunk 2]
```diff
@@ -31,12 +32,14 @@ type getMsgFunc func(tCtx *testContext, preData interface{}) sdk.Msg
...
 
-type verificationFunc func(t *testing.T, tCtx *testContext, result sdk.Result, testCase testCaseItem, preCoins, afterCoins sdk.SysCoins, preData interface{})
+type verificationFunc func(t *testing.T, tCtx *testContext, err sdk.Error, testCase testCaseItem, preCoins, afterCoins sdk.SysCoins, preData interface{})
 
-var verification verificationFunc = func(t *testing.T, context *testContext, result sdk.Result, testCase testCaseItem, preCoins, afterCoins sdk.SysCoins, preData interface{}) {
-	if result.Log == "" && testCase.expectedCode != sdk.CodeOK || result.Log != "" && testCase.expectedCode == sdk.CodeOK {
-		fmt.Println(result.Log)
-		require.Zero(t, 1)
+var verification verificationFunc = func(t *testing.T, context *testContext, err sdk.Error, testCase testCaseItem, preCoins, afterCoins sdk.SysCoins, preData interface{}) {
+	if err == nil {
```
Comment (ilovers) [inline x/farm/handler_test.go]: should replace err to testCase. expectedErr

File: x/farm/handler_test.go [hunk 3]
```diff
@@ -234,7 +237,7 @@ func TestHandlerMsgCreatePool(t *testing.T) {
 			preExec:      preExec,
 			getMsg:       normalGetCreatePoolMsg,
 			verification: verification,
-			expectedCode: sdk.CodeOK,
+			expectedCode: sdk.EnvelopedErr{Err: nil},
```
Comment (ilovers) [inline x/farm/handler_test.go]: replace `sdk.EnvelopedErr{Err: nil}` to nil.

File: x/farm/types/proposal_test.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ func TestNewManageWhiteListProposal(t *testing.T) {
 		description string
 		poolName    string
 		isAdded     bool
-		errCode     sdk.CodeType
+		errCode     uint32
```
Comment (ilovers) [inline x/farm/types/proposal_test.go]: do not need change this.

**Review by summerpro — APPROVED** [review]

**Review by ilovers — APPROVED** [review]

**Pull request merged**